### PR TITLE
refactor and unit tests for exporter

### PR DESCRIPTION
**Pull request by pvighi**
Changes:
* added unit tests for exporter lambda
* fixed error handling
* refactored

Right now we only have one  successful fulfilment test case taken from CODE. But we can very easily add more test cases just by adding the zuora query csvs and the expected resulting fulfilment file. We should add more cases for tricky edge cases as we discover them.
@AWare

### CODE REVIEW

File: src/exporter.js [hunk 1]
```diff
@@ -64,101 +45,144 @@ export function handler (input, context, callback) {
...
+    if (deliveryDate.isValid()) {
+      resolve(deliveryDate)
+    } else {
+      reject(new Error('invalid deliverydate expected format YYYY-MM-DD'))
+    }
+  })
+}
+
+async function asyncHandler (input) {
+  let stage = await getStage()
```
Comment (AWare) [inline src/exporter.js]: Why not await config and then just use the state field.

File: src/exporter.js [hunk 2]
```diff
@@ -1,19 +1,14 @@
-import AWS from 'aws-sdk'
 import csv from 'fast-csv'
```
Comment (AWare) [inline src/exporter.js]: It would be nice to get flow on this file.
 `// @flow`

File: src/exporter.js (hunk 1 above)
Comment (pvighi) [inline src/exporter.js]: I don't really need to download the config from s3 for this lambda but It might make sense to do some useless processing and downloading just to keep it consistent with the other lambdas

File: src/exporter.js (hunk 2 above)
Comment (pvighi) [inline src/exporter.js]: oops I keep forgetting about that, I'll add it
